src/ext/java/org/opentripplanner/ext/transmodelapi/TransmodelAPI.java

```diff
@@ -141,16 +174,34 @@ public Response getGraphQLBatch(List<HashMap<String, Object>> queries, @HeaderPa
             }
             String operationName = (String) query.getOrDefault("operationName", null);
 
-            futures.add(() -> index.getGraphQLExecutionResult((String) query.get("query"), router,
-                    variables, operationName, maxResolves));
+            futures.add(() -> index.getGraphQLExecutionResult(
+                (String) query.get("query"),
+                router,
+                variables,
+                operationName,
+                maxResolves,
+                getTagsFromHeaders(headers)
+            ));
         }
 
         try {
             List<Future<ExecutionResult>> results = index.threadPool.invokeAll(futures);
-            return Response.status(Response.Status.OK).entity(GraphQLResponseSerializer.serializeBatch(queries, results)).build();
+            return Response
+                .status(Response.Status.OK)
+                .entity(GraphQLResponseSerializer.serializeBatch(queries, results))
+                .build();
         } catch (InterruptedException e) {
             LOG.error("Batch query interrupted", e);
             throw new RuntimeException(e);
         }
     }
+
+    private static Iterable<Tag> getTagsFromHeaders(HttpHeaders headers) {
+        return tracingHeaderTags.stream()
+            .map(header -> {
+                String value = headers.getHeaderString(header);
+                return Tag.of(header, value == null ? "__UNKNOWN__" : value);
+            })
+            .collect(Collectors.toList());
+    }
```

Comment:
Question - Do we want all headers to tags? I think this will cause every request to be stored in the `Meter` for the duration of the pull intervals. F.eks the set of `Content-NNN` tags will be unique for a given request. I think it is better to white-list the tags we want in the configuration. Thoughts on this?